lldb/source/Plugins/TypeSystem/Swift/TypeSystemSwiftTypeRef.cpp

```diff
@@ -1758,8 +1758,13 @@ void TypeSystemSwiftTypeRef::DumpSummary(opaque_compiler_type_t type,
 }
 bool TypeSystemSwiftTypeRef::IsPointerOrReferenceType(
     opaque_compiler_type_t type, CompilerType *pointee_type) {
-  return m_swift_ast_context->IsPointerOrReferenceType(ReconstructType(type),
-                                                       pointee_type);
+  auto impl = [&]() {
+    return
+      IsPointerType(type, pointee_type) ||
+      IsReferenceType(type, pointee_type, nullptr);
+  };
+  VALIDATE_AND_RETURN(impl, IsPointerOrReferenceType, type,
+                      (ReconstructType(type), pointee_type));
```

Comment:
That looks obviously correct — do you think it's still worth adding a quick unit test to `lldb/unittest/Symbol/TestTypeSystemSwiftTypeRef.cpp`?

Reply:
Instead of adding a new test case, I augmented the existing pointer and reference tests to call `IsPointerOrReferenceType` with the types they create for checking `IsPointerType` and `IsReferenceType` respectively. Does this seem reasonable to you too?